src/realm/list.cpp

```diff
@@ -136,58 +136,55 @@ void Lst<T>::sort(std::vector<size_t>& indices, bool ascending) const
 {
     update_if_needed();
 
-    if constexpr (std::is_same_v<T, Mixed>) {
-        if (ascending) {
-            do_sort(indices, size(), [this](size_t i1, size_t i2) {
-                return get(i1) < get(i2);
-            });
-        }
-        else {
-            do_sort(indices, size(), [this](size_t i1, size_t i2) {
-                return get(i1) > get(i2);
-            });
-        }
+    auto tree = m_tree.get();
+    if (ascending) {
+        do_sort(indices, size(), [tree](size_t i1, size_t i2) {
+            return unresolved_to_null(tree->get(i1)) < unresolved_to_null(tree->get(i2));
+        });
     }
     else {
-        auto tree = m_tree.get();
-        if (ascending) {
-            do_sort(indices, size(), [tree](size_t i1, size_t i2) {
-                return tree->get(i1) < tree->get(i2);
-            });
-        }
-        else {
-            do_sort(indices, size(), [tree](size_t i1, size_t i2) {
-                return tree->get(i1) > tree->get(i2);
-            });
-        }
+        do_sort(indices, size(), [tree](size_t i1, size_t i2) {
+            return unresolved_to_null(tree->get(i1)) > unresolved_to_null(tree->get(i2));
+        });
     }
 }
 
+// std::unique, but leaving the minimum value rather than the first found value
+// for runs of duplicates. This makes distinct stable without relying on a
+// stable sort, which makes it easier to write tests and avoids surprising results
+// where distinct appears to change the order of elements
+template <class Iterator, class Predicate>
+static Iterator min_unique(Iterator first, Iterator last, Predicate pred)
+{
```

Comment:
I'd recommend `REALM_ASSERT_DEBUG(first != last)` here

Reply:
I made it just return first if first == last to match std::unique.